bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/helpers/RepositoryHelper.java

```diff
@@ -149,18 +148,19 @@ public static <T> IRepository<T> createMemoryComposite(IProvisioningAgent agent,
 	 * the Eclipse Installer.
 	 *
 	 * @return an unmodifiable list of global shared bundle pools.
-	 *
-	 * @throws IOException if there are problems loading the pool information.
 	 */
 	@SuppressWarnings("nls")
-	public static List<Path> getSharedBundlePools() throws IOException {
-			Path bundlePools = Path.of(System.getProperty("user.home"), ".p2/pools.info");
-			if (Files.isRegularFile(bundlePools)) {
-				try (Stream<String> lines = Files.lines(bundlePools, getSharedBundlePoolEncoding())) {
-					return lines.map(Path::of).filter(Files::isDirectory).toList();
-				}
+	public static List<Path> getSharedBundlePools() {
+		Path bundlePools = Path.of(System.getProperty("user.home"), ".p2/pools.info");
+		if (Files.isRegularFile(bundlePools)) {
+			try (Stream<String> lines = Files.lines(bundlePools, getSharedBundlePoolEncoding())) {
+				return lines.map(Path::of).filter(Files::isDirectory).toList();
+			} catch (Exception ex) {
+				LogHelper.log(
+						new Status(IStatus.WARNING, Activator.ID, "The bundle pool load failed: " + bundlePools, ex));
```

Comment:
```suggestion
						Status.warning("The bundle pool load failed: " + bundlePools, ex));
```
would be a bit simpler. :)

Reply:
`ILog.error("The bundle pool load failed: " + bundlePools, ex);` would even be simpler ;-)

Reply:
> `ILog.warn("The bundle pool load failed: " + bundlePools, ex);` would even be simpler ;-)

I think it needs to be `ILog.get().warn(...`, but I agree that there is a lot of potential in P2 to improve logging (and other things).^^

Reply:
What was the comment?  Perfect is the enemy of good.  😈